src/LanguageServer/Impl/Sources/DefinitionSource.cs

```diff
@@ -43,34 +44,46 @@ internal sealed class DefinitionSource {
             var eval = analysis.ExpressionEvaluator;
             using (eval.OpenScope(analysis.Document, exprScope)) {
                 var value = await eval.GetValueFromExpressionAsync(expr, cancellationToken);
-                return await FromMemberAsync(value, expr, eval, cancellationToken);
+                return await FromMemberAsync(value, expr, statement, analysis, cancellationToken);
             }
         }
 
-        private async Task<Reference> FromMemberAsync(IMember value, Expression expr, IExpressionEvaluator eval, CancellationToken cancellationToken) {
+        private async Task<Reference> FromMemberAsync(IMember value, Expression expr, Node statement, IDocumentAnalysis analysis, CancellationToken cancellationToken) {
             cancellationToken.ThrowIfCancellationRequested();
 
             Node node = null;
             IPythonModule module = null;
+            LocationInfo location = null;
+            var eval = analysis.ExpressionEvaluator;
 
             switch (value) {
                 case IPythonClassType cls:
                     node = cls.ClassDefinition;
                     module = cls.DeclaringModule;
+                    location = cls.Location;
                     break;
                 case IPythonFunctionType fn:
                     node = fn.FunctionDefinition;
                     module = fn.DeclaringModule;
+                    location = fn.Location;
                     break;
                 case IPythonPropertyType prop:
                     node = prop.FunctionDefinition;
                     module = prop.DeclaringModule;
+                    location = prop.Location;
                     break;
                 case IPythonModule mod: {
                         var member = eval.LookupNameInScopes(mod.Name, out var scope);
```

Comment:
Can we extract this whole case into separate method?

Reply:
Sure